### ISSUE TITLE
Help using constraints with topLayoutGuide

### ISSUE THREAD

**Issue by eduardo4jesus** [stefanrenne/ConstraintLib#1]
I am trying to use `self.topLayoutGuide` to position my widget. But I can't compile. It says 

/Users/eduardoj/Qwantech/Blupper_Iphone/Blupper/Blupper/BlupRevealVC.swift:28:21: Type of expression is ambiguous without more context

Here is my code:

```
let newView = UIView()
mainStack.addSubview(newView)
newView.pin([.top: self.topLayoutGuide, .left:0.0, .height:100.0, .width:100.0])
newView.backgroundColor = UIColor.red
```

I already tried:

`newView.pin([.top: self.topLayoutGuide.length, .left:0.0, .height:100.0, .width:100.0])
`
and 

`newView.pin([.topTo(self.topLayoutGuide): 0, .left:0.0, .height:100.0, .width:100.0])
`
but didn't work.

**Comment by stefanrenne** [comment]
Hi eduardo4jesus,

First of all, thank you for using my lib. I really appreciate it.

Now lets dive into the `topLayoutGuide.length` property.
This function returns a CGFloat and you need to pass in a Float (welcome to the Objective-c inheritance).

Using your code as an example, you could solve this in one of the following ways:

```
let newView = UIView() 
mainStack.addSubview(newView)
newView.pin([.top: Float(self.topLayoutGuide), .left:0.0, .height:100.0, .width:100.0])
newView.backgroundColor = UIColor.red
```

or

```newView.pin([.top: Float(self.topLayoutGuide.length), .left:0.0, .height:100.0, .width:100.0])```

By the way:
`.topTo(View)` was meant to link one view to another view.

Example: 
```view4.pin([.left: 100.0, .right: 100.0, .height: 30.0, .topTo(view2): 30.0])```
Also take a look at the demo project.

I should mention that the LayoutGuide's length value is dynamically updated. As example when the in call status bar is being displayed. I will think of a solution to add a constraint to the LayoutGuide's

Happy coding and best regards,

Stefan Renne

**Comment by stefanrenne** [comment]
Hi eduardo4jesus,

I just release version 1.0.4. 
In this version you can create a relation to the viewController's layout guide's, allowing a view to always stick to the status bar and even the in-call status bar. Example:

```
let topLayoutGuideConstraint = newView1.pin(.topLayoutGuide, constant: 0.0)
let bottomLayoutGuideConstraint = newView1.pin(.bottomLayoutGuide, constant: 0.0)
```

Or of course one of the other formats.
```
let topLayoutGuideConstraint = newView1.pin(.topLayoutGuide)
newView1.pin([.topLayoutGuide, .bottomLayoutGuide, .left, .right]
newView1.pin([.topLayoutGuide: 0.0, .bottomLayoutGuide: 0.0, .left: 0.0, .right: 0.0]
```

Side note, it is not yet possible to search for a `topLayoutGuide` relation. The following code will therefore not yet work, this is scheduled for v1.0.5.

```
let topLayoutGuideConstraint = view1.constraint(.topLayoutGuide)
let bottomLayoutGuideConstraint = view1.constraint(.bottomLayoutGuide)
```

**Comment by eduardo4jesus** [comment]
Hi stefanrenne, thank you so much for supporting.

I really like using this notation
`newView1.pin([.topLayoutGuide: 0.0, .bottomLayoutGuide: 0.0, .left: 0.0, .right: 0.0]`

Is there a way to change the multiplier as well as the constant?

Best Regards,

Eduardo Reis

**Comment by stefanrenne** [comment]
Thank you for your reply, I guess I could add the multiplier, but I think it would then be something like this:
```
let topConstraint = newView1.pin(.top, constant: 50.0, multiplier: 1.0)
```

Would you mind if the multiplier approach is not available in the dictionary approach?

Stefan

**Comment by stefanrenne** [comment]
Hi eduardo4jesus,

I just released 1.0.5 with the following documentation:




Example 6: Default multiplier

The default multiplier can only be modified in the single line creator
```
let newView1 = UIView()
self.view.addSubview(newView1)
let topConstraint = newView1.pin(.top, constant: 50.0, multiplier: 2.0)
```
After initialisation, the multiplier can be modified by using the multiplier attribute
```
newView1.constraint(.top)?.multiplier = 1.5
```

**Comment by eduardo4jesus** [comment]
Hi stefanrenne,

> Would you mind if the multiplier approach is not available in the dictionary approach?

I don't. I can't think in a nice way to have  a dictionary single line approach for it, like 
`view.pin([.left: 100.0, .right: 100.0, .height: 30.0, .topTo(view2): 30.0])`.

Thanks for the quick updates on the library. I really appreciated.
